packages/camera/camera_android_camerax/lib/src/android_camera_camerax.dart

```diff
@@ -423,7 +423,7 @@ class AndroidCameraCameraX extends CameraPlatform {
   /// [cameraId] not used.
   @override
   Future<void> pausePreview(int cameraId) async {
-    _unbindUseCaseFromLifecycle(preview!);
+    await _unbindUseCaseFromLifecycle(preview!);
     _previewIsPaused = true;
```

Comment:
This looks like the mistake I made in maps: you're making a field update asynchronously instead of synchronously. You probably want to reverse the statement order.

Reply:
Ah okay yes I'll reverse the order!

Reply:
This was also the case in `resumePreview`, so I modified it there, too.